docs/src/main/asciidoc/WildFly_Maven_Plugin_Guide.adoc

```diff
@@ -0,0 +1,410 @@
+[[WildFly_Maven_Plugin_Guide]]
+= WildFly Maven Plugin Guide
+WildFly team;
+:revnumber: {version}
+:revdate: {localdate}
+:toc: macro
+:toclevels: 3
+:toc-title: WildFly Maven Plugin Guide
+:doctype: book
+:icons: font
+:source-highlighter: coderay
+
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+// ifndef::ebook-format[:leveloffset: 1]
+
+(C) 2024 The original authors.
+
+ifdef::basebackend-html[toc::[]]
+:numbered:
+
+[[wildfly_maven_plugin]]
+= WildFly Maven Plugin
+
+This document details the steps to follow in order to develop a WildFly application
+using the https://docs.wildfly.org/wildfly-maven-plugin[WildFly Maven Plugin].
+
+Developing an application is not different from developing an application for a traditional
+WildFly server installation using Maven.
+The extra steps required to provision the WildFly server and deploy your application inside the server are handled by the 
+https://docs.wildfly.org/wildfly-maven-plugin[_org.wildfly.plugins:wildfly-maven-plugin_] Maven plugin `package` goal.
```

Comment:
It seems we should mention `image` and `provision` here as well.

Reply:
I introduced the goal in the introductory chapter.